trillian/integration/hammer.go

```diff
@@ -478,51 +667,101 @@ func (s *hammerState) getSTH(ctx context.Context) error {
 	return nil
 }
 
-func (s *hammerState) getSTHConsistency(ctx context.Context) error {
+// chooseSTHs gets thje current STH, and also picks an earlier STH.
+func (s *hammerState) chooseSTHs(ctx context.Context) (*ct.SignedTreeHead, *ct.SignedTreeHead, error) {
 	// Get current size, and pick an earlier size
 	sthNow, err := s.client().GetSTH(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to get-sth for current tree: %v", err)
+		return nil, nil, fmt.Errorf("failed to get-sth for current tree: %v", err)
 	}
 	which := rand.Intn(sthCount)
 	if s.sth[which] == nil {
 		glog.V(3).Infof("%s: skipping get-sth-consistency as no earlier STH", s.cfg.LogCfg.Prefix)
 		s.needOps(ctfe.GetSTHName)
-		return errSkip{}
+		return nil, nil, errSkip{}
 	}
 	if s.sth[which].TreeSize == 0 {
 		glog.V(3).Infof("%s: skipping get-sth-consistency as no earlier STH", s.cfg.LogCfg.Prefix)
 		s.needOps(ctfe.AddChainName, ctfe.GetSTHName)
-		return errSkip{}
+		return nil, nil, errSkip{}
 	}
 	if s.sth[which].TreeSize == sthNow.TreeSize {
 		glog.V(3).Infof("%s: skipping get-sth-consistency as same size (%d)", s.cfg.LogCfg.Prefix, sthNow.TreeSize)
 		s.needOps(ctfe.AddChainName, ctfe.GetSTHName)
-		return errSkip{}
+		return nil, nil, errSkip{}
+	}
+	return s.sth[which], sthNow, nil
+}
+
+func (s *hammerState) getSTHConsistency(ctx context.Context) error {
+	sthOld, sthNow, err := s.chooseSTHs(ctx)
+	if err != nil {
+		return err
 	}
 
-	proof, err := s.client().GetSTHConsistency(ctx, s.sth[which].TreeSize, sthNow.TreeSize)
+	proof, err := s.client().GetSTHConsistency(ctx, sthOld.TreeSize, sthNow.TreeSize)
 	if err != nil {
-		return fmt.Errorf("failed to get-sth-consistency(%d, %d): %v", s.sth[which].TreeSize, sthNow.TreeSize, err)
+		return fmt.Errorf("failed to get-sth-consistency(%d, %d): %v", sthOld.TreeSize, sthNow.TreeSize, err)
 	}
-	if err := checkCTConsistencyProof(s.sth[which], sthNow, proof); err != nil {
-		return fmt.Errorf("get-sth-consistency(%d, %d) proof check failed: %v", s.sth[which].TreeSize, sthNow.TreeSize, err)
+	if err := checkCTConsistencyProof(sthOld, sthNow, proof); err != nil {
+		return fmt.Errorf("get-sth-consistency(%d, %d) proof check failed: %v", sthOld.TreeSize, sthNow.TreeSize, err)
 	}
 	glog.V(2).Infof("%s: Got STH consistency proof (size=%d => %d) len %d",
-		s.cfg.LogCfg.Prefix, s.sth[which].TreeSize, sthNow.TreeSize, len(proof))
+		s.cfg.LogCfg.Prefix, sthOld.TreeSize, sthNow.TreeSize, len(proof))
 	return nil
 }
 
 func (s *hammerState) getSTHConsistencyInvalid(ctx context.Context) error {
 	if s.lastTreeSize() == 0 {
-		return nil
+		return errSkip{}
+	}
+
+	choices := []Choice{ParamTooBig, ParamsInverted, ParamNegative, ParamInvalid}
+	choice := choices[rand.Intn(len(choices))]
+
+	var err error
+	var proof [][]byte
+	switch choice {
+	case ParamTooBig:
+		first := s.lastTreeSize() + uint64(invalidStretch)
+		second := first + 100
```

Comment:
Looks like this covers both first and second being too big - what about if only second is too big, and first is less than the current tree size?  Could be another invalidity case?

Feel free to ignore this suggestion for now though :)

Reply:
Done.